lib/client-assets.php

```diff
@@ -1139,7 +1139,7 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	}
 
 	if ( ! empty( $color_palette ) ) {
-		$editor_settings['colors'] = $color_palette;
+		$editor_settings['colors'] = apply_filters( 'editor_color_palette', $color_palette );
```

Comment:
This is cool, thanks for applying changes to this PR. In my opinion, it reads better on JS side and removes some unnecessary complexity.

Just wanted to double check, if we really need another filter in here? Maybe it would be enough to call this function directly and defer filter when there are requests from plugin developers. Otherwise, we will have to document it and what's worse maintain forever. It seems like `get_theme_support( 'editor-color-palette' )` should offer enough control. This change is needed mostly for backward compatibility.

Reply:
IMHO there really can't be enough filters. This gives plugins and child-themes the chance to extend/modify an existing set of colors. Together with a class like [phpColors](http://mexitek.github.io/phpColors/) I can imagine the possibilities to be amazing.

Reply:
I'm not sure if we should have this filter right now or not, it is another artifact to maintain, and even if we don't add it now, we can add it later. Also if we add this filter here, we may need for consistency reasons to add filters to other theme settings.
Another discussion to have on this kind of filters is if they should be PHP or client-side JavaScript filters. The client side filters offer the advantage of allowing plugins to have UI that dynamically change the values during Gutenberg execution. It would be possible to vary color palette depending on post size, post format or tags for example.

If we decide to use the filter I feel it should be called even if the color palette was not set (with an empty array) so filters can add colors. After filter execution, we verify if the colors are not empty and in that case, we set the editor colors if they are not empty we don't set them (so no bytes travel and Gutenberg uses the default colors).

Reply:
Should we remove that filter and move that discussion to a later date?

Reply:
Yes, please. Let's open a separate issue for that.